deepchem/models/dft/nnxc.py

```diff
@@ -0,0 +1,152 @@
+from abc import abstractproperty, abstractmethod
+from typing import Union
+try:
+    from dqc.xc.base_xc import BaseXC
+    from dqc.utils.datastruct import ValGrad, SpinParam
+    from dqc.api.getxc import get_xc
+    import torch
+except ModuleNotFoundError:
+    raise ModuleNotFoundError("This layer requires dqc and torch")
+
+
+class BaseNNXC(BaseXC, torch.nn.Module):
+    """
+    Base class for the NNLDA and HybridXC classes.
+    """
+
+    @abstractproperty
+    def family(self) -> int:
+        pass
+
+    @abstractmethod
+    def get_edensityxc(
+            self, densinfo: Union[ValGrad, SpinParam[ValGrad]]) -> torch.Tensor:
+        pass
+
+
+class NNLDA(BaseNNXC):
+    """
+    Neural network xc functional of LDA (only receives the density as input)
```

Comment:
Can you add a usage example? There should be an explanation of what LDA is as a DFT method, along with a suitable reference. Aim for at least a few paragraphs of discussion